### PR TITLE
fix correctness issue with 'ok' binding

### PR DESCRIPTION
**Pull request by MHBauer**
courtesy of @arschles. Hopefully I didn't make this mistake elsewhere. Please be on the lookout.

### CODE REVIEW

**Comment by duglin** [comment]
still don't understand why we need to check the 'old' ones, but if we're going to check then LGTM

File: pkg/registry/servicecatalog/instance/strategy.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func (instanceRESTStrategy) ValidateUpdate(ctx kapi.Context, new, old runtime.Ob
 	if !ok {
 		glog.Warning("received a non-instance object to validate to")
 	}
-	oldInstance := old.(*sc.Instance)
+	oldInstance, ok := old.(*sc.Instance)
```
Comment (arschles) [inline pkg/registry/servicecatalog/instance/strategy.go]: not blocking for this PR, but if `ok` is false, then `oldInstance` will be `nil`

**Comment by MHBauer** [comment]
we also decided that we should exit on a failure to assert type, so changing the logging from Warning to Fatal, will achieve that goal.

**Comment by MHBauer** [comment]
@arschles @duglin re-review needed.

**Comment by duglin** [comment]
still LGTM just please squash

**Pull request merged**